lua/bson_lua_wrapper.c

```diff
@@ -0,0 +1,372 @@
+#include <stdlib.h>
+#include <string.h>
+ 
+#include <lua5.2/lua.h>
+#include <lua5.2/lauxlib.h>
+ 
+#include <bson_object.h>
+
+static int table_to_bson_object(lua_State *L, BsonObject *obj, char *errorMessage);
+static int table_to_bson_array(lua_State *L, BsonArray *arr, char *errorMessage);
+
+static int bson_object_to_table(lua_State *L, BsonObject *obj, char *errorMessage);
+static int bson_element_to_table(lua_State *L, BsonElement *element, char *errorMessage);
+static int bson_array_to_table(lua_State *L, BsonArray *arr, char *errorMessage);
+
+static int lbson_object_to_bytes(lua_State *L) {
+    BsonObject bson;
+    bson_object_initialize_default(&bson);
+    if (!lua_istable(L, 1)) {
+        return luaL_argerror(L, 1, "Arg 1 must be a table");
+    }
+
+    char errorMessage[255];
+    int success = table_to_bson_object(L, &bson, errorMessage);
+    if (success != 0) {
+        bson_object_deinitialize(&bson);
+        return luaL_argerror(L, 1, errorMessage);
+    }
+
+    uint8_t *bytes = bson_object_to_bytes(&bson);
+    if (bytes == NULL) {
```

Comment:
@jacobkeeler I would prefer pull consts (NULL) on left side of compaction. And make bytes const

Reply:
The issue in this case is that this memory is allocated. The `free` function takes a non-const `void *` argument, and a warning is shown if the pointer that is passed in is `const`. This can be casted away in theory, but that seems like bad practice as well. This admittedly is a bit of a grey area.